src/DaedalusCompiler/Compilation/SemanticErrors.cs

```diff
@@ -51,30 +56,48 @@ public class AttributeNotFoundException : Exception
         
     }
 
-    public class ErrorContext
+    public abstract class BaseErrorContext
     {
-        public ParserRuleContext Context;
+        public ParserRuleContext ParserContext;
         public string FilePath;
-        public string[] FileContentLines;
+    }
+
+    public class ErrorLineContext : BaseErrorContext
+    {
+        public readonly string FileContentLine;
+
+        public ErrorLineContext(ErrorFileContext errorFileContext)
+        {
+            ParserContext = errorFileContext.ParserContext;
+            FilePath = errorFileContext.FilePath;
+
+            FileContentLine = errorFileContext.FileContentLines[errorFileContext.ParserContext.Start.Line - 1];
```

Comment:
Why you initialize that with `errorFileContext.FileContentLines[errorFileContext.ParserContext.Start.Line - 1]` ?

Reply:
It's because we don't preserve FileContentLines and when we start parsing new file, old FileContentLines are deleted, but we somehow need to store line when the symbol occured (to use it later), that's why it's like that.